ui/src/app/dashboard/dashboard.component.spec.ts

```diff
@@ -67,9 +70,55 @@ describe('DashboardComponent', () => {
     expect(testComponent).toBeTruthy();
   }));
 
+  it("should create expected amount of cards", fakeAsync(() => {
+    let matCards = fixture.debugElement.queryAll(By.css('mat-card'));
+    expect(matCards.length).toEqual(CARD_NUM);
+  }));
+
+  it('should navigate to job-list page (fake) when status counts are clicked', fakeAsync(() => {
+    let countAnchor = fixture.debugElement.query(By.css("jm-dashboard a")).nativeElement;
+    countAnchor.click();
+    fixture.detectChanges();
+    tick();
+    expect(fixture.debugElement.query(By.css("div")).nativeElement.textContent).toEqual("fake job-list page");
+  }));
+
+  it('should has status as url param when totalSummaryComponent links are clicked', fakeAsync(() => {
+    let totalSummaryAnchor = fixture.debugElement.query(By.css("jm-total-summary .count a")).nativeElement;
+    let status = fixture.debugElement.query(By.css("jm-total-summary .status")).nativeElement.textContent.trim();
+
+    totalSummaryAnchor.click();
+    fixture.detectChanges();
+    tick();
+    testJobListComponent = fixture.debugElement.query(By.css("jm-test-job-list-component")).componentInstance;
+    expect(testJobListComponent.route.snapshot.queryParams['q'].indexOf(status)).toBeGreaterThan(-1);
+  }));
+
+  it("should has status and label as url params when groupedSummaryComponent links are clicked", fakeAsync(() => {
+    let groupedSummaryAnchor = fixture.debugElement.query(By.css("jm-grouped-summary tr td.count a")).nativeElement;
+    groupedSummaryAnchor.click();
+    tick();
+    testJobListComponent = fixture.debugElement.query(By.css("jm-test-job-list-component")).componentInstance;
+    let q = testJobListComponent.route.snapshot.queryParams['q'];
```

Comment:
Same as above, would parse the search request out instead, and check a specific field.